### DISCUSSION TITLE
Collapsible Code Blocks in Code Editor

### DISCUSSION THREAD

**Discussion by Parkertg** [standardnotes/forum#2740] [Suggestions]
**Have you read the [Longevity Statement](https://standardnotes.org/longevity)?** yes

**Describe the solution you'd like**
Using the code editor, I was disappointed to see that there is not option to collapse various code structures, like a function or class etc.
From what I can tell, codeMirror already supports this, but it would just need to be setup? Correct me if I'm wrong!

**Alternative that only works for readonly situations (in Markdown editor)**
<details>
  <summary>Click to expand</summary>
  whatever
</details>

**Additional Thoughts to the feature request**
I saw someone else request an outline editor. and besides actually working with code blocks, I enjoy taking notes with sections that can be minimized etc. Using Notepad++ there's an option to add custom "Languages" in which you can specify how the editor handles various symbols. you can specify that for example "##" means the beginning of a collapsible zone and "~-" is the end of the section..  Which I find to be handy when making notes. So ability to add 

**Links/Possible how to**
[stackoverflow folding solution](https://stackoverflow.com/questions/8309835/xml-code-code-folding-or-expand-collapse-functionality-using-codemirror)
I'm still a bit too new here to understand how to build and host this myself. Maybe in the future I could try to do it myself. will take some additional learning

**Comment by JaspalSuri** [comment]
Hi @Parkertg, thanks for the suggestion and additional feedback. Adding code folding to the Code Editor is on our todo list. 🙂
